sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.service.ts

```diff
@@ -119,6 +127,128 @@ export class SIMSToSFASService {
+   */
+  async getAllApplicationsWithUpdates(
+    modifiedSince: Date,
+    modifiedUntil: Date,
+  ): Promise<ApplicationRecord[]> {
+    return (
+      this.applicationRepo
+        .createQueryBuilder("application")
+        .select("application.id", "applicationId")
+        .addSelect("student.id", "studentId")
+        .addSelect("programYear.programYear", "programYear")
+        .addSelect("offering.offeringIntensity", "offeringIntensity")
+        // Use CASE to conditionally select studyStartDate and studyEndDate and casting JSON values to dates.
+        .addSelect(
+          `CASE
+      WHEN studentAssessment.offering IS NULL THEN (application.data->>'studyStartDate')::date
+      ELSE offering.study_start_date
+     END`,
+          "studyStartDate",
+        )
+        .addSelect(
+          `CASE
+      WHEN studentAssessment.offering IS NULL THEN (application.data->>'studyEndDate')::date
+      ELSE offering.study_end_date
+     END`,
+          "studyEndDate",
+        )
+        // Summing CSGP awards where value_code is 'CSGP'.
+        .addSelect(
+          `SUM(CASE WHEN disbursementValues.value_code = 'CSGP' THEN disbursementValues.valueAmount ELSE 0 END)`,
+          "csgpAwardTotal",
+        )
+        // Summing SBSD awards where value_code is 'SBSD.
+        .addSelect(
+          `SUM(CASE WHEN disbursementValues.value_code = 'SBSD' THEN disbursementValues.valueAmount ELSE 0 END)`,
+          "sbsdAwardTotal",
+        )
+        // Application cancel date when status is cancelled.
+        .addSelect(
+          `CASE
+      WHEN application.applicationStatus = :cancelled THEN application.application_status_updated_on
+      ELSE NULL
+     END`,
+          "applicationCancelDate",
+        )
+        .innerJoin("application.currentAssessment", "studentAssessment")
+        .innerJoin("application.programYear", "programYear")
+        .innerJoin("application.student", "student")
+        .innerJoin("studentAssessment.offering", "offering")
+        .leftJoin(
+          "studentAssessment.disbursementSchedules",
+          "disbursementSchedule",
+        )
+        .leftJoin(
+          "disbursementSchedule.disbursementValues",
+          "disbursementValues",
+        )
+        .where("application.applicationStatus != :overwritten")
+        // Check if the application data was updated in the given period.
+        .andWhere(
+          "application.updatedAt > :modifiedSince AND application.updatedAt <= :modifiedUntil",
+        )
+        .setParameters({
+          overwritten: ApplicationStatus.Overwritten,
+          cancelled: ApplicationStatus.Cancelled,
+          modifiedSince,
+          modifiedUntil,
+        })
+        .groupBy("application.id")
+        .addGroupBy("student.id")
+        .addGroupBy("programYear.id")
+        .addGroupBy("studentAssessment.id")
+        .addGroupBy("offering.id")
+        .orderBy("offering.offeringIntensity")
```

Comment:
We are ordering by offering intensity and as there is no specific order mentioned by business this should be fine.